### PR TITLE
Add a cache options to reduce latency

### PR DESCRIPTION
**Pull request by eilgin**
This will allow projects to improve their responsiveness.
 
I've taken the opportunity of this PR to refactor some parts of the file (mainly using `async`/`await`) 😉

Btw, it's a great service to work with!

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -3,76 +3,76 @@ declare module 'flagsmith-nodejs' {
      * Initialise the sdk against a particular environment
      */
     export function init(config: {
-        environmentID: string
-        onError?: Function
-        defaultFlags?: string[]
-        api?: string
-    }): void
+        environmentID: string;
```
Comment (kyle-ssg) [inline index.d.ts:6]: I believe this would need a new entry for the cache option

**Review by kyle-ssg — COMMENTED** [review]
Awesome! Just a minor comment on types

File: index.d.ts (hunk 1 above)
Comment (eilgin) [reply index.d.ts:6]: Right! I'll fix that

Comment (kyle-ssg) [reply index.d.ts:6]: Great, cheers so much for this!

**Comment by eilgin** [comment]
I've also edited the docs: https://github.com/Flagsmith/flagsmith-docs/pull/11

**Comment by kyle-ssg** [comment]
@eilgin published under flagsmith-nodejs@1.1.0

**Pull request merged**